python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -771,6 +771,19 @@ def my_dbt_op(dbt: DbtCliResource):
             manifest, dagster_dbt_translator = get_manifest_and_translator_from_dbt_assets(
                 [assets_def]
             )
+
+            dbt_resource_props_by_test_name = get_dbt_resource_props_by_test_name(manifest)
+            has_asset_check = any(
+                dbt_resource_props.get("meta", {}).get("dagster", {}).get("asset_check", False)
+                for dbt_resource_props in dbt_resource_props_by_test_name.values()
+            )
```

Comment:
Is my understanding correct here that toggling a single test as an asset check can impact the selection behavior of other tests (that aren't asset checks) within the dbt project?

This seems like potentially confusing behavior. Have we considered the alternative of forcing the "load tests as checks" decision to be made at the project level, instead of for individual tests?

Reply:
I agree that we should add that constraint. I don't think we can enforce that in the dbt level though, but when we load the manifest in Dagster when they use `@dbt_assets` or `load_assets_from_dbt_XXX`, we can assert that all tests have this metadata, if one test has it set.

Reply:
https://github.com/dagster-io/dagster/pull/16852